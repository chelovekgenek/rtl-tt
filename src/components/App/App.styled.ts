import styled, { createGlobalStyle } from "styled-components"

export const Global = createGlobalStyle`
  body {
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
      'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
      sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`

export const Layout = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`
