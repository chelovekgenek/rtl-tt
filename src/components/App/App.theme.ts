export default {
  colors: {
    primary: "#ff9201",
  },
}
