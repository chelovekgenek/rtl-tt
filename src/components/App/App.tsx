import React from "react"
import { Provider } from "react-redux"
import { ThemeProvider } from "styled-components"

import { Header } from "components/Header"
import { store } from "store"

import * as Styles from "./App.styled"
import theme from "./App.theme"
import { Articles } from "components/Articles"

export const App = () => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <Styles.Global />
      <Styles.Layout>
        <Header />
        <Articles />
      </Styles.Layout>
    </Provider>
  </ThemeProvider>
)
