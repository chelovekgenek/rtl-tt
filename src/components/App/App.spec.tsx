import React from "react"
import { shallow } from "enzyme"

import { App } from "./App"

describe("App", () => {
  const render = (props: Partial<React.ComponentProps<typeof App>> = {}) => shallow(<App {...props} />)

  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
})
