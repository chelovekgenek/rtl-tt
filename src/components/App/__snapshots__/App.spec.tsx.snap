// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "primary": "#ff9201",
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <GlobalStyleComponent />
    <styled.div>
      <Connect(Header) />
      <Connect(Articles) />
    </styled.div>
  </Provider>
</ThemeProvider>
`;
